### ISSUE TITLE
Cluster List Search performs badly at scale

### ISSUE THREAD

**Issue by richard-cox** [rancher/dashboard#11994]
**Setup**
- Rancher version:  2.9.1 (and before)

**Describe the bug**
![image](https://github.com/user-attachments/assets/fc6621ce-3e1b-43d6-bc94-2935b15af751)

- 17 second churn when tryng to filter a list of ~700 clusters in the cluster list home page
- functions are all generic / anon, no way to track down what's causing the CPU churn aside from reproducing locally

**To Reproduce**
- Load ~700 clusters (they could be RKE2 Custom)
- nav to home page
- enter text

**Result**
- not much of anything...

**Expected Result**
- list should promptly display result of filter

**Additional context**
- with vai on by default this should go away
- tracking here in case something happens or we need to get in something before

**Comment by richard-cox** [comment]
Resolved by https://github.com/rancher/dashboard/issues/10283

Should only be tested once https://github.com/rancher/dashboard/issues/12734 is resolved

**Comment by IsaSih** [comment]
so, issue #12734 is still open. SHould I move this issue to QA blocked?

**Comment by richard-cox** [comment]
Blocked on https://github.com/rancher/dashboard/issues/10283

**Comment by IsaSih** [comment]
As per screen recording https://github.com/user-attachments/assets/c652101c-e6f4-4056-a6b8-47d754eaf56f 
the cluster list crashes on both Home page and Clusters management while the clusters are still being created. This is not ideal behavior. @richard-cox should this be addressed in a new issue or should i reopen this one?
Repro steps:
1- Imported the yaml with 700 clusters
rancher version:


[Rancher](https://github.com/rancher/rancher)	v2.15.0-alpha17
[Dashboard](https://github.com/rancher/dashboard)	master 7570de1
[Rancher CLI](https://github.com/rancher/cli)	v2.15.0-rc.1

**Comment by richard-cox** [comment]
if the issue is about performance at scale, reopen this one. If not would need to see the image/video (the link is broken). Would definitely be interested in more detail and steps to reproduce, this isn't something seen before.

**Comment by izaac** [comment]
moved back to review, I moved i by mistake

**Comment by IsaSih** [comment]
Works as expected. Verified in 

[Rancher](https://github.com/rancher/rancher)	v2.15.0-alpha17
[Dashboard](https://github.com/rancher/dashboard)	master 7570de1
[Rancher CLI](https://github.com/rancher/cli)	v2.15.0-rc.1

Average loading time is 130 ms in both clusters management and home page lists

Clusters management:
https://github.com/user-attachments/assets/833e9390-bbb0-4588-915c-9832db67b454

Home page:

https://github.com/user-attachments/assets/724fdaaa-4fa8-42af-a10a-48ec07aa1dae

**Comment by IsaSih** [comment]
I'll retest the lists with still loading clusters once #18344 is merged

**Comment by izaac** [comment]
# Performance validation: cluster list filter at scale 

Verdict: reproduced on 2.14, confirmed fixed on 2.15.

Filtering the Home page cluster list with 700 clusters was measured on Rancher 2.14.3-rc1 (no fix) and 2.15.0-alpha21 (fix present). The regression reproduces clearly on 2.14 and is resolved on 2.15 by PR [#17234](https://github.com/rancher/dashboard/pull/17234), which moves the Home and Cluster Management lists to management.cattle.io with server-side pagination. That PR is on the v2.15.0 milestone and seems not backported to 2.14.

## Headline

Home page, filter query that matches all 700 clusters, first run after seeding:

| Version | Result | Notes |
| :-- | :-- | :-- |
| 2.14.3-rc1 (no fix) | 42.2 s | worst observed 60+ s, main thread churns |
| 2.15.0-alpha21 (fix) | 1.18 s | 0.40 s once the sql-cache is warm |

About 35x faster cold, about 150x faster warm.

<img width="2040" height="1800" alt="Image" src="https://github.com/user-attachments/assets/08aba5ee-31dc-4ad7-bfd5-c5e676f40c88" />

## Full results (milliseconds)

| Page / operation | 2.14.3-rc1 | 2.15.0-alpha21 |
| :-- | --: | --: |
| Home, filter matches all 700 (cold) | 42230 | 1176 |
| Home, filter matches all 700 (warm) | 60230 | 402 |
| Home, filter narrow match (~10 rows) | 218 | 334 |
| Home, first page load | 476 | 461 |
| Cluster Management, filter matches all 700 | 408 | 518 |

The Home list is the regression. On 2.14 it renders client-side and is not wired to server-side pagination, so a large match set churns for tens of seconds, and the ui-performance setting does not change that. The Cluster Management list already used a server-paginated store and was fast on both versions. On 2.15 the Home list is server-paginated and stays responsive.

<details>
<summary>Environment and method</summary>

Both versions were brought up locally on k3d with default settings and seeded with 700 imported provisioning.cattle.io clusters via kubectl. Each imported cluster also creates a management.cattle.io cluster, so both lists render 700 rows.

Timing was captured in-page with performance.now() plus a MutationObserver so a blocked main thread is still measured, and a PerformanceObserver for long tasks. Load is time to a quiet DOM after a fresh navigation. Filter is time from the keystroke to a settled row set. Each case ran 3 iterations. The cold first run is reported as the honest number because 2.15 uses a sql-cache that is still populating on the first pass.

Pins: 2.14 = rancher-latest 2.14.3 (resolved 2.14.3-rc1), 2.15 = rancher-alpha 2.15.0-alpha21.

</details>
